document_page_portal/controllers/portal.py

```diff
@@ -0,0 +1,116 @@
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+
+from odoo import http
+from odoo.exceptions import AccessError, MissingError
+from odoo.http import request
+from odoo.tools.translate import _
+from odoo.addons.portal.controllers.portal import pager as portal_pager, CustomerPortal
+from odoo.osv.expression import OR
+
+
+class CustomerPortal(CustomerPortal):
+
+    def _prepare_portal_layout_values(self):
+        values = super(CustomerPortal, self)._prepare_portal_layout_values()
+        values['document_page_count'] = request.env[
+            'document.page'].search_count([('type', '=', 'content')])
+        return values
+
+    def _document_page_get_page_view_values(self, document_page,
+                                            access_token, **kwargs):
+        values = {
+            'page_name': 'document_page',
+            'document_page': document_page,
+        }
+        return self._get_page_view_values(
+            document_page, access_token, values,
+            'my_document_pages_history', False, **kwargs)
+
+    @http.route(
+        ['/my/document_pages', '/my/document_pages/page/<int:page>'],
+        type='http', auth="user", website=True)
+    def portal_my_knowledge_document_pages(self, page=1, date_begin=None,
+                                           date_end=None, sortby=None,
+                                           search=None, search_in='content', **kw):
+        values = self._prepare_portal_layout_values()
+        # user = request.env.user
+        domain = [('type', '=', 'content')]
+
+        searchbar_sortings = {
+            'date': {'label': _('Newest'), 'order': 'create_date desc'},
+            'name': {'label': _('Name'), 'order': 'name'},
+            'parent': {'label': _('Category'), 'order': 'parent_id'},
+        }
+        searchbar_inputs = {
+            'content': {'input': 'content', 'label': _(
+                'Search <span class="nolabel"> (in Content)</span>')},
+            'all': {'input': 'all', 'label': _('Search in All')},
+        }
+
+        # default sort by value
+        if not sortby:
+            sortby = 'date'
+        order = searchbar_sortings[sortby]['order']
+
+        # archive groups - Default Group By 'create_date'
+        archive_groups = self._get_archive_groups('document.page', domain)
+        if date_begin and date_end:
+            domain += [('create_date', '>', date_begin), (
+                'create_date', '<=', date_end)]
```

Comment:
```suggestion
            domain += [
                ('create_date', '>', date_begin),
                ('create_date', '<=', date_end),
            ]
```

not blocking but IMO writing domains like this is much more readable